furbot.rb

```diff
@@ -197,6 +197,58 @@
     msg.channel.send("https://duckduckgo.com/?q=#{CGI.escape(term)}")
   end
 
+  on :message, /!log (\A[a-z|A-Z|\d|.|_|-|#|,]{1,20}\z)$/ do |msg, logsearchtext1|
+    # Showing first newest irclog search result
+    logresultcounter1 = 0
+    logfilelist1 = Dir['/logs/plainlogs/*.log'].sort.reverse
+    logfilelist1.each do |logfilename1|
+       File.readlines(logfilename1).reverse_each do |filelogline1|
+          completelogline1 = "#{logfilename1[-14..-5]} #{filelogline1}"
+          if (completelogline1[logsearchtext1])
+             logresultcounter1 = logresultcounter1 + 1
+             msg.channel.send("#{completelogline1}")
+             exit
+          end
+       end
+    end
+    if logresultcounter1 == 0 then
+       msg.channel.send("Not found.")
```

Comment:
It’s preferred to use `msg.reply`, because that’s independant of the way furbot received the message. If someone uses `/query` on furbot, he will respond properly with a PM in that case again, while using `msg.channel.send` will always result in a public message in the channel, if any. If the message has no channel (PM), your code will crash here, as `msg.channel` will be `nil`.